### DISCUSSION TITLE
News since 20220722

### DISCUSSION THREAD

**Discussion by dixyes** [swow/swow#127] [Ideas]
开发进展记录

# v0.2.x迁移

v0.1.x基本上功能冻结，接下来会发布比较正式的版本，但这个版本上不提供功能更新，只修比较好修的bug

purposed/v0.2.x上进行新版本的开发，准备完成后将会合入ci分支

预期 11月中旬进行这两件事

v0.2.x上的进展见 #136

# v0.2.x migration

v0.1.x features have been frozen, we will make a release version, but this version won't get any feature updates, only minor bug fixes provided.

We will develop 0.2.x on purposed/v0.2.x branch, when it get ready, we will merge it into ci branch.

Estimate date: mid November

For v0.2.x news, see #136

## 修复

* 修复使用第三方包实现 psr7 Response时返回 OK问题 （#128）
* 修复了杀掉一个正在退出的协程时错误的行为
* 修复了协程bailout行为

## 增加 删除 修改

+ 添加了`Message->addHeader(string $name, mixed $value): static`和`Message->unsetHeader(string $name): static`方法

- 移动php-cs-fixer的配置到单独的仓库

* 重构了一些测试

(记录随时变更)

## Fix

* Fixed behavior when using a custom psr-7 implement (#128)
* Fixed unexpected behavior on killing a exiting coroutine
* Fixed Coroutine bailout behavior

## Addition Removal Change

+ Add `Message->addHeader(string $name, mixed $value): static` and `Message->unsetHeader(string $name): static` methods

- Move php-cs-fixer config to separate repository

* Refactored some tests

(this may change at anytime)

**Comment by AuroraYolo** [comment]
听君一席话，胜似一席话

**Reply by x-rush** [reply]
犹如庄周带净化。
